templates/experiment_wrapper.html

```diff
@@ -99,20 +111,42 @@
                     }else{
                         alert(failed_ajax + ": 1");
                     }
-                    _record_task_complete = true;
-					//$(".expt-pop-outer").fadeOut("slow");
-                    _hide_loader();
                 },
                 error: function (jqXHR, textStatus, errorThrown){
                     alert(_failed_ajax + ": 2");
-                    _record_task_complete = true;
-					//$(".expt-pop-outer").fadeOut("slow");
+                },
+                complete: function(jqXHR, textStatus ){
                     _hide_loader();
+                    if(_rep===1&&_rep_on_consent===0){
+                        //Need to delay _record_task_complete until after
+                        //_mark_rep_as_done finished if REP and not on consent (i.e., credit at end of experiment)
+                        _mark_rep_as_done();
+                    }else{
+                        _record_task_complete = true;
+                    }
+					//$(".expt-pop-outer").fadeOut("slow");
+                }
+            });
+        };
+        var _mark_rep_as_done = function(){
+            $.ajax({
+                type: 'GET',
+                url: '/mark-rep-as-done?survey_code=' + _survey_code + '&uid=' + _uid,
+				//async: false,
+                success: function (data) {
+                        //alert("Your REP has been marked as done");
+                },
+                error: function (jqXHR, textStatus, errorThrown){
+                    alert("Your REP has not been updated,Here is the error message:\n"+jqXHR.responseText);
```

Comment:
@benston3 Ben actually the current build already does what you described. If the mark_rep_as_done() method in python returns a 500 code plus the error message. _mark_rep_as_done method in JS will catch it and give the user an alert. The user will still be able to continue with the experiment(just tested it).

Reply:
Okay, excellent